### PR TITLE
fix leading whitespace, add test for nested exception formatting

### PR DESCRIPTION
**Pull request by 6palace**

### CODE REVIEW

File: src/main/java/com/ge/predix/log4j1/PredixLayoutPattern.java [hunk 1]
```diff
@@ -41,11 +44,7 @@ protected String convert(final LoggingEvent event) {
         }
         logFormat.put("msg", event.getMessage());
         if (null != event.getThrowableInformation()) {
```
Comment (sanjeevchopra) [inline src/main/java/com/ge/predix/log4j1/PredixLayoutPattern.java]: this null check can be removed

File: src/main/java/com/ge/predix/log4j1/PredixLayoutPattern.java [hunk 2]
```diff
@@ -54,4 +53,22 @@ protected String convert(final LoggingEvent event) {
...
+    
+    private List<List<String>> getStackTrace(final LoggingEvent event) {
+        List<List<String>> exceptions = new ArrayList<>();
+        Throwable throwable = event.getThrowableInformation().getThrowable();
+        while (throwable != null) {
+            List<String> stack = new ArrayList<>();
+            stack.add(throwable.getClass().getName()
+                    + (!StringUtils.isEmpty(throwable.getMessage()) ? ": " + throwable.getMessage() : ""));
+            for (StackTraceElement element : throwable.getStackTrace()) {
+                stack.add("at " + element.toString());
```
Comment (sanjeevchopra) [inline src/main/java/com/ge/predix/log4j1/PredixLayoutPattern.java]: add 4 spaces before at for indents?

File: src/test/java/com/ge/predix/log4j1/PredixLayoutTest.java [hunk 3]
```diff
@@ -143,11 +143,11 @@ public void testPredixLayoutExceptionChainLog() throws IOException {
                 + "\",\"inst\":\"" + INSTANCE_ID_VALUE + "\",\"tid\":\"" + THREAD_NAME + "\",\"mod\":\"" + FILE_NAME
                 + "\",\"lvl\":\"" + Level.ERROR.toString()
                 + "\",\"msg\":\"{width=4, length=3, units=inches, height=5}\",\"stck\":["
-                + "\"java.lang.Exception: java.lang.NullPointerException\","
+                + "[\"java.lang.Exception: java.lang.NullPointerException\","
```
Comment (sanjeevchopra) [inline src/test/java/com/ge/predix/log4j1/PredixLayoutTest.java]: pls update the test to use a message in both the root and cause exceptions.

File: src/main/java/com/ge/predix/log4j1/PredixLayoutPattern.java (hunk 1 above)
Comment (6palace) [inline src/main/java/com/ge/predix/log4j1/PredixLayoutPattern.java]: I think the null check is important for non-error logs, where we don't want a stck property in the json at all. I can replace with a ternary operator, but the principle would remain the same.

File: src/main/java/com/ge/predix/log4j1/PredixLayoutPattern.java (hunk 2 above)
Comment (6palace) [inline src/main/java/com/ge/predix/log4j1/PredixLayoutPattern.java]: Good idea, will do.

Comment (6palace) [inline src/main/java/com/ge/predix/log4j1/PredixLayoutPattern.java]: The json structure with 4 spaces indent is just as confusing as without indenting, maybe not?

**Pull request merged**